sgt-back-end/index.js

```diff
@@ -0,0 +1,130 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+app.get('/api/grades/', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades/', (req, res) => {
+  if (!req.body.name || !req.body.course || !req.body.score) {
+    res.status(400).json({ error: 'name, course, and score are required fields' });
+    return;
+  } else if (req.body.score < 1 || req.body.score > 100) {
+    res.status(400).json({ error: 'score must be an integer between 1-100' });
+    return;
+  }
+
+  const sql = `
+      insert into "grades" ("name", "course", "score")
+      values ($1, $2, $3)
+      returning *;
+  `;
+  const params = [req.body.name, req.body.course, req.body.score];
+  const newGrade = req.body;
+
+  db.query(sql, params)
+    .then(result => {
+      res.status(201).json(newGrade);
```

Comment:
The client should receive the row including the id and timestamp assigned by the database (`result.rows[0]`).